bin/eds-import-from-figma.js

```diff
@@ -0,0 +1,258 @@
+
+  // eslint-disable-next-line import/extensions
+  const { hideBin } = require('yargs/helpers');
+  const { prompt } = require('enquirer');
+  const at = require('lodash/at');
+  const set = require('lodash/set');
+
+  const { default: ora } = await import('ora');
+  const chalk = require('chalk');
+
+  // Set up the command structure to output help if the file to load is not specified
+  const args = yargs(hideBin(process.argv))
+    .command('$0 <file>', 'Figma variable file to import from', (yargs) => {
+      yargs.positional('file', {
+        describe: 'JSON tokens file emitted from the plugin',
+        type: 'string',
+      });
+    })
+    .option('verbose', {
+      describe: 'Print additional details for debugging purposes',
+      type: 'boolean',
+    })
+    .option('dry-run', {
+      describe: 'Run the script without writing any changes',
+      type: 'boolean',
+    }).argv;
+
+  // read in the config from config file, package json "eds", etc.
+  const { getConfig } = require('./_util');
+  const config = await getConfig();
+
+  const isVerbose = args.v;
+  const canWrite = !args.dryRun;
+
+  // Read the figma import file, and parse out the modes to select from, prompting the user to pick one
+  let importTheme;
+  try {
+    importTheme = JSON.parse(fs.readFileSync(args.file, 'utf8'));
+  } catch (e) {
+    console.error(chalk.red('cannot load import file', args.file));
+    process.exit(1);
+  }
+
+  // now, load in the local theme file. We want to look up keys in there
+  let localTheme;
+  try {
+    localTheme = JSON.parse(
+      fs.readFileSync(`${config.src}app-theme.json`, 'utf8'),
+    );
+  } catch (e) {
+    console.error(
+      chalk.red('cannot load local theme file', `${config.src}app-theme.json`),
+    );
+    process.exit(2);
+  }
+
+  // Get the file being used: [primitives, themes]
+  let tokenFile;
+  switch (importTheme.name.toLowerCase()) {
+    case 'themes':
+      tokenFile = 'themes';
+      break;
+    case 'primitives':
+      tokenFile = 'primitives';
+      break;
+    default:
+      tokenFile = 'unknown';
+  }
+
+  // Determine which of the modes in the file should be used
+  let response;
+  try {
+    response = await prompt({
+      name: 'mode',
+      message: 'Please select the Figma mode for token import',
+      type: 'select',
+      choices: Object.values(importTheme.modes),
+    });
+  } catch (e) {
+    // e.g., someone hits ESC
+    console.error(chalk.red('Aborted.'));
+    process.exit(-1);
+  }
+
+  // Find the mode, and get the stored key for later use
+  const [modeKey] = Object.entries(importTheme.modes).filter(
+    (modeTuple) => modeTuple[1] === response.mode,
+  )[0];
+
+  /**
+   * The actual parsing
+   * TODO: separate all of the below out into separate file(s)
+   */
+  const stats = {
+    skipped: [],
+    updated: [],
+    errored: [],
+  };
+
+  const spinner = ora('Parsing tokens').start();
+  importTheme.variables.forEach((figmaVariable) => {
```

Comment:
This is basically the "main loop" and may be factored out into some more generic module at some point. mostly does a few things

Given a token / variable from figma... 

* format the token name for code
* format the path to write to
* convert the format of the token value (using the earlier selected mode)

Reply:
This is the "main loop", is there value is pulling this code out to test it?

Reply:
Yah for sure. what will likely happen is that at some point in the future, the output from figma will change and break some/all of this. Having tests will mean we can update the data used in those instead of poking around the whole script to make updates